### PR TITLE
[csharp]: Deep Link Consuming Subentity Id for Tabs

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
### Description

This sample app demonstrates deep link consuming subentity id in Static Tabs and Configurable Tabs.
Language Used: C#

**Proposed Changes**
1. Static Tabs deeplink consuming sub entity id
2. Configurable Tabs deeplink from chat to Tab consuming subentity id

### Testing

Bot post a card in chat:
image

Static Tabs
image

Channel Tab
image

### CODE REVIEW

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/tab-deeplink/csharp/DeepLinkBot/DeepLinkHelperChannel.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace Microsoft.BotBuilderSamples.Bots
+{
+    public class DeepLinkHelperChannel
+    {
+        public static string channelID = DeepLinkBot.channelID;
+
+        static string task1Values =>  task1Json();
+        static string task1Json()
+        {
```
Comment (Wajeed-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/DeepLinkHelperChannel.cs:12]: Can we create function which takes parameters and return deeplink string? 

public string GetDeeplinkToTab(string appId, string entityId, string subEntitiyId, string channelId)

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/deeplink-subentityId/csharp/DeepLinkBot/Views/Shared/_EmbedPage.cshtml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+<html lang="en">
+<head>
+    <meta name="viewport" content="width=device-width" />
+    <title>@ViewBag.Title</title>
+    <script src="https://statics.teams.cdn.office.net/sdk/v1.5.2/js/MicrosoftTeams.min.js" asp-append-version="true"></script>
+
+    @RenderSection("scripts", required: false)
+</head>
+<body>
+    <style>
```
[the remark this answers is not in the document]
Comment (Trinetra-MSFT) [reply samples/deeplink-subentityId/csharp/DeepLinkBot/Views/Shared/_EmbedPage.cshtml]: Removed this file

File: samples/deeplink-subentityId/csharp/DeepLinkBot/Views/Shared/_ValidationScriptsPartial.cshtml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿<environment names="Development">
+    <script src="~/lib/jquery-validation/dist/jquery.validate.js"></script>
```
[the remark this answers is not in the document]
Comment (Trinetra-MSFT) [reply samples/deeplink-subentityId/csharp/DeepLinkBot/Views/Shared/_ValidationScriptsPartial.cshtml]: Removed this file

File: samples/tab-deeplink/csharp/DeepLinkBot/DeepLinkHelperChannel.cs (hunk 1 above)
Comment (Trinetra-MSFT) [reply samples/tab-deeplink/csharp/DeepLinkBot/DeepLinkHelperChannel.cs:30]: Created method but we need to send subentity ID for every deeplink on click adaptive card button

File: samples/tab-deeplink/csharp/DeepLinkBot/DeeplinkHelper.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    public class DeeplinkHelper
+    {
+        public static IConfiguration _configuration;
+        public DeeplinkHelper(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+        static Dictionary<string, string> task1Values = new Dictionary<string, string>
+            {
+                {"subEntityId","topic1" }
```
Comment (Wajeed-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/DeeplinkHelper.cs:17]: Have two methods to fetch deeplink. 

string GetDeepLinkToPersonalTab()
string  GetDeepLinkToChannelTab()

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

**Review by Wajeed-msft — APPROVED** [review]

**Review by asnow003 — APPROVED** [review]

**Pull request merged**